R/col_plan.R

```diff
@@ -343,27 +355,47 @@ select_col_plan <- function(data, tfrmt){
         ) %>%
         relocate(.data$.original_col, .after = last_col()) %>%
         relocate(.data$.rename_col, .after = last_col()) %>%
-        unite("new_name_in_df", -.rename_col , sep = .tlang_delim, remove = FALSE, na.rm = TRUE) %>%
-        unite("new_name_in_df_output", c(-.original_col, -new_name_in_df), sep = .tlang_delim, remove = FALSE, na.rm = TRUE) %>%
-        mutate(new_name_quo = map(.data$new_name_in_df, sym)) %>%
+        unite("new_name_in_df", c(-.data$.rename_col, -.data$.removal_identifier_col) , sep = .tlang_delim, remove = FALSE) %>%
+        unite("new_name_in_df_output", c(-.data$.original_col, -.data$new_name_in_df, -.data$.removal_identifier_col), sep = .tlang_delim, remove = FALSE) %>%
         mutate(
-          new_name_in_df = case_when(
-            new_name_in_df == .original_col ~ NA_character_,
-            TRUE ~ new_name_in_df
-          )
-        )
+          new_name_in_df = remove_empty_layers(.data$new_name_in_df, n_layers),
+          new_name_in_df_output = remove_empty_layers(.data$new_name_in_df_output, n_layers)
+        ) %>%
+        mutate(
+          new_name_quo = map2(.data$new_name_in_df, .data$.removal_identifier_col, function(x, y) {
```

Comment:
Given this already an anonymous function can you make it a named function and pull it out this easier to maintain

Reply:
NP. will do.